### ISSUE TITLE
Unexplained errors

### ISSUE THREAD

**Issue by MSoup** [exercism/dart#349]
Hi there. Apart from hello world, every problem I have attempted so far results in an error that doesn't provide guidance on where to look. For example, here is my code for two fer:

```
String twoFer([String? name = 'you']) {
  return "One for ${name}, one for me.";
}
```

Apart from any glaring flaws I might have in my code (or do I have any?), the test tab shows:

```
An error occurred while running your tests. This might mean that there was an issue in our infrastructure, or it might mean that you have something in your code that's causing our systems to break.

Please check your code, and if nothing seems to be wrong, try running the tests again.
```

Is there something I'm doing wrong? On my CLI, my tests pass, but it seems like my submission is incorrect when I test it on the site.

**Comment by Stargator** [comment]
This seems like an issue related to https://github.com/exercism/dart-test-runner/issues/26

The test runner will say an error has occurred, but we don't know where the source of the issue is.

@iHiD or @ErikSchierboom  Would one fo you be able to look at the logs and see more details on the issue?

**Comment by iHiD** [comment]
@MSoup Sorry about this and thanks for the tag @Stargator.

I'm unsure on this one. We get output (your tests passed) but a 512 with no error that I can see, which means we show the error. I'll have to look into it more. I just tried on hello-world and it looked like it timed out. @ErikSchierboom Could you look at this too pls. I'll try and get to it once I've done a few other urgent issues. Thanks.

**Comment by Stargator** [comment]
@iHiD I just tested the dart-test-runner against a solution for leap. The build.log is fine, but the results.json has:

```json
{
  "version": 1,
  "status": "fail",
  "message": "Failed to precompile test:test:\n../../../../../.pub-cache/hosted/pub.dartlang.org/http_multi_server-2.1.0/lib/src/multi_headers.dart:The method 'MultiHeaders.add' has fewer named arguments than those of overridden method 'HttpHeaders.add'.\n  void add(String name, Object value) {\n       ^\norg-dartlang-sdk:///sdk/lib/_http/http.dart:This is the overridden method ('add').\n  void add(String name, Object value,\n       ^\n../../../../../.pub-cache/hosted/pub.dartlang.org/http_multi_server-2.1.0/lib/src/multi_headers.dart:The method 'MultiHeaders.set' has fewer named arguments than those of overridden method 'HttpHeaders.set'.\n  void set(String name, Object value) {\n       ^\norg-dartlang-sdk:///sdk/lib/_http/http.dart:This is the overridden method ('set').\n  void set(String name, Object value,\n       ^"
}
```

I think I have a solution for the test-runner (the above issue) and it may be related to this. I tested a solution for rna-transcription, but there was no issue there.

For the issue shown in the results.json above, that is from Dart trying to use older dependencies for a build that needs newer dependencies. I'm not sure if the dart-test-runner caches dependencies or is supposed to get fresh dependencies. If the latter, then it could be that the `pubspec.lock` file is not being removed and that's why the older dependencies are being retrieved.

I am fairly confident that the issue reported by @MSoup and https://github.com/exercism/dart-test-runner/issues/26 are related and belong to the dart-test-runner.

But if the logs can give us a clearer picture, then we can wait before deciding whether to move this issue.

**Comment by Stargator** [comment]
I have a possible solution in the works in https://github.com/exercism/dart-test-runner/pull/28

**Comment by Stargator** [comment]
https://github.com/exercism/dart-test-runner/pull/28 Issue seemed similar and has resolved itself, could you try again and let me know what happens?

**Comment by MSoup** [comment]
Small update:

It looks like by removing the ? in my function via the advice of a mentor (dont use null safety??), the test still failed the first time, but resubmitting it a few days later shows that the tests passed. This gave rise to a new bug, which I'll comment about in a bit.

**Comment by Stargator** [comment]
Okay, we just pushed an update to how the tests are executed, is it still passing?

**Comment by MSoup** [comment]
Yes, the test passes. When there is a hiccup, however, the test fails with unexplained errors, such as this just now:

<img width="913" alt="Screen Shot 2021-09-07 at 23 23 14" src="https://user-images.githubusercontent.com/19704779/132361829-7689ec41-7c1f-4119-b029-736095dfb5a0.png">

(I assume there's something wrong with my code, though there's no way I can tell when using the online editor)

**Comment by iHiD** [comment]
It's timing out and being killed after 10s of running.

**Comment by Stargator** [comment]
@iHiD okay so this is a test runner issue?

**Comment by iHiD** [comment]
@Stargator Yes, I think it is. I can get you more info from the logs if you need/want.

**Comment by Stargator** [comment]
@iHiD Sure that would be good.
